### PR TITLE
use dbus instead of notify-send

### PR DESCRIPTION
**Pull request by googol42**
- use dbus instead of notify-send. See here: https://github.com/albertlauncher/python/pull/85#issuecomment-723681735
- also use plugin icon in notification

### CODE REVIEW

**Comment by googol42** [comment]
How are the plugins updates? With regular releases?

**Review by ManuelSchneid3r — COMMENTED** [review]
> How are the plugins updates? With regular releases?

yes

File: timer/__init__.py [hunk 1]
```diff
@@ -12,20 +12,26 @@
...
 from threading import Timer
 from time import strftime, time, localtime
+import dbus
 import os
+import random
 import subprocess
 
 __iid__ = "PythonInterface/v0.1"
 __prettyname__ = "Timer"
 __version__ = "1.1"
```
[the remark this answers is not in the document]
Comment (googol42) [reply timer/__init__.py]: sure

File: timer/__init__.py [hunk 2]
```diff
@@ -35,11 +41,13 @@ def timeout():
...
-                subprocess.Popen(['notify-send', 'Timer "%s"' % self.name, '-t', '0', subtext])
-            else:
-                subprocess.Popen(['notify-send', 'Timer', '-t', '0', subtext])
+
+            title = 'Timer "%s"' % self.name if self.name else 'Timer'
+            text = "Timed out at %s" % strftime("%X", localtime(self.end))
+
+            bus = dbus.SessionBus()
+            notify = dbus.Interface(bus.get_object(dbusItem, dbusPath), dbusInterface)
+            notify.Notify(__prettyname__, random.randint(0, 100000), iconPath, title, text, '', '', 0)
```
Comment (ManuelSchneid3r) [inline timer/__init__.py:50]: I see no point in this random. If possible use None otherwise send 0. No need to draw a random REPLACE ID.

**Review by ManuelSchneid3r — CHANGES_REQUESTED** [review]

File: timer/__init__.py [hunk 3]
```diff
@@ -12,20 +12,26 @@
...
 import os
+import random
 import subprocess
 
 __iid__ = "PythonInterface/v0.1"
 __prettyname__ = "Timer"
-__version__ = "1.1"
+__version__ = "1.2"
 __trigger__ = "timer "
 __author__ = "Manuel Schneider, Andreas Preikschat"
```
Comment (ManuelSchneid3r) [inline timer/__init__.py:24]: Please use your github username for easy mentions. When you're at it please change mine to manuelschneid3r as well.

File: timer/__init__.py (hunk 2 above)
Comment (ManuelSchneid3r) [inline timer/__init__.py:50]: I see no point in this random. If possible use None otherwise send 0. No need to draw a random REPLACE ID.

**Review by ManuelSchneid3r — CHANGES_REQUESTED** [review]

File: timer/__init__.py (hunk 2 above)
Comment (googol42) [reply timer/__init__.py]: done

File: timer/__init__.py (hunk 3 above)
Comment (googol42) [reply timer/__init__.py]: done

File: timer/__init__.py [hunk 4]
```diff
@@ -35,11 +40,13 @@ def timeout():
...
-                subprocess.Popen(['notify-send', 'Timer "%s"' % self.name, '-t', '0', subtext])
-            else:
-                subprocess.Popen(['notify-send', 'Timer', '-t', '0', subtext])
+
+            title = 'Timer "%s"' % self.name if self.name else 'Timer'
+            text = "Timed out at %s" % strftime("%X", localtime(self.end))
+
+            bus = dbus.SessionBus()
+            notify = dbus.Interface(bus.get_object(dbusItem, dbusPath), dbusInterface)
+            notify.Notify(__prettyname__, 0, iconPath, title, text, '', '', 0)
```
Comment (ManuelSchneid3r) [inline timer/__init__.py:49]: didnt 'None' work?

**Review by ManuelSchneid3r — APPROVED** [review]

File: timer/__init__.py (hunk 4 above)
Comment (googol42) [reply timer/__init__.py:49]: I didn't try `None`

**Pull request merged**